crates/validation/src/order/state/db_state_utils/balances.rs

```diff
@@ -55,18 +57,36 @@ impl Balances {
         user: Address,
         token: Address,
         db: &DB
-    ) -> Option<U256>
+    ) -> U256
     where
         <DB as DatabaseRef>::Error: Debug + Sync + Send + 'static
     {
-        self.0
+        // Fetch balance from the token contract
+        let token_balance = self
+            .tokens
             .get(&token)
             .or_else(|| {
                 let slot = find_slot_offset_for_balance(db, token);
                 let slot = TokenBalanceSlot::new(token, slot as u8);
-                self.0.insert(token, slot);
-                self.0.get(&token)
+                self.tokens.insert(token, slot);
+                self.tokens.get(&token)
             })
             .and_then(|slot| slot.load_balance(user, db).ok())
+            .unwrap_or_default();
+
+        let angstrom_balance = self.get_angstrom_balance(token, user, db);
+        token_balance + angstrom_balance
```

Comment:
So this isn't correct. Because we only pull from internal balances if its flagged in the order itself. this can lead to big issues and cause reverts / failures in the system.

Reply:
Right. Let me fix that.